### DISCUSSION TITLE
Customize marker with Composable

### DISCUSSION THREAD

**Discussion by siyingch86** [patrykandpatrick/vico#648] [Questions] [answered]
Hi, I want to custom a marker like the following picture 
<img width="264" alt="Screenshot 2024-04-05 at 8 32 05 AM" src="https://github.com/patrykandpatrick/vico/assets/160801134/32900c20-5c80-4fb1-875b-e8192b21318e">

I'm using Vico 2.0 and Compose. Wanna know if there's any way to extend the "Marker" interface and use a composable to present the bubble marker. Thanks!

**Comment by Gowsky** [comment, marked as answer]
Hi @siyingch86, there's currently no way to use composable functions for `Marker`. However, you can implement any kind of marker by drawing on the `Canvas` using Vico or Android APIs. For your custom marker, you can modify [`MarkerCorneredShape`](https://github.com/patrykandpatrick/vico/blob/a274cfb968b56e73a23677fa4d1a49f2cbd3a7e1/vico/core/src/main/java/com/patrykandpatrick/vico/core/component/shape/cornered/MarkerCorneredShape.kt) and use it along with `TextComponent`.

We'll be looking into supporting composable functions for markers in the future.
